### PR TITLE
Re add Add shopping based on recipes

### PR DESCRIPTION
**Pull request by lhellebr**
Original PR: https://github.com/instructlab/taxonomy/pull/330

### CODE REVIEW

File: writing/freeform/shopping/qna.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+- answer: You need to buy salt, yeast and mozzarella.
+  question: I'm making pizza. I need flour, water, salt, yeast, tomato puree, mozzarella and basil. I have flour, water, tomato puree and basil. What do I need to buy?
+- answer: You need to buy salt.
```
Comment (bmr-cymru) [inline writing/freeform/shopping/qna.yaml:4]: Shouldn't this be flour, if you already have water and salt?

File: writing/freeform/shopping/qna.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  question: I want to bake flatbread. The recipe says I need just water, salt and flour. I have water and salt, what do I need to buy?
+- answer: You don't need to buy anything!
+  question: I wish to make pizza for dinner. The recipe saus I need flour, water, salt, yeast, tomato puree, mozzarella and basil. I have water, salt, flour, mozzarella, basil and tomato puree. What should I buy?
```
Comment (bmr-cymru) [inline writing/freeform/shopping/qna.yaml:7]: s/saus/says/

File: writing/freeform/shopping/qna.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+---
+- answer: You need to buy salt, yeast and mozzarella.
+  question: I'm making pizza. I need flour, water, salt, yeast, tomato puree, mozzarella and basil. I have flour, water, tomato puree and basil. What do I need to buy?
+- answer: You need to buy salt.
+  question: I want to bake flatbread. The recipe says I need just water, salt and flour. I have water and salt, what do I need to buy?
+- answer: You don't need to buy anything!
```
Comment (bmr-cymru) [inline writing/freeform/shopping/qna.yaml:6]: Aren't you missing yeast?

File: writing/freeform/shopping/qna.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  question: I wish to make pizza for dinner. The recipe saus I need flour, water, salt, yeast, tomato puree, mozzarella and basil. I have water, salt, flour, mozzarella, basil and tomato puree. What should I buy?
+- answer: You need to buy salt, water and flour.
+  question: I want to make flatbread. I heard you only need water, salt and flour but I don't have any of these. What should I buy?
+- answer: Yes, you need to buy sugar.
+  question: "I'm making lemonade. Recipe says: water, lemon, sugar. I have water and lemon, do I need to buy anything?"
+- answer: Yes, you need to buy sugar and salt.
+  question: "I'm making lemonade. Recipe says: water, lemon, sugar and a pinch of salt. I have water and lemon, do I need to buy anything?"
+- answer: You need to buy mozzarella or some other cheese of your choice.
+  question: "I'm making pizza. The recipe: flour, water, salt, yest, tomato puree, mozzarella or other cheese and a handful of basil. I have flour, water, tomato puree and basil. What do I need to buy?"
+- answer: You need to buy mozzarella or some other cheese of your choice, and basil.
```
Comment (bmr-cymru) [inline writing/freeform/shopping/qna.yaml:16]: Missing yeast?

**Review by bmr-cymru — CHANGES_REQUESTED** [review]

**Comment by lhellebr** [comment]
Thanks for the catches, I'd make for a terrible AI :)
Fixed.

File: writing/freeform/shopping/qna.yaml (hunk 3 above)
Comment (ckadner) [reply writing/freeform/shopping/qna.yaml:6]: This one seems resolved

File: writing/freeform/shopping/qna.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  question: I want to bake flatbread. The recipe says I need just water, salt and flour. I have water and salt, what do I need to buy?
+- answer: You don't need to buy anything!
+  question: I wish to make pizza for dinner. The recipe saus I need flour, water, salt, yeast, tomato puree, mozzarella and basil. I have water, salt, flour, mozzarella, basil and tomato puree. What should I buy?
+- answer: You need to buy salt, water and flour.
+  question: I want to make flatbread. I heard you only need water, salt and flour but I don't have any of these. What should I buy?
+- answer: Yes, you need to buy sugar.
+  question: "I'm making lemonade. Recipe says: water, lemon, sugar. I have water and lemon, do I need to buy anything?"
+- answer: Yes, you need to buy sugar and salt.
+  question: "I'm making lemonade. Recipe says: water, lemon, sugar and a pinch of salt. I have water and lemon, do I need to buy anything?"
+- answer: You need to buy mozzarella or some other cheese of your choice.
```
[the remark this answers is not in the document]
Comment (ckadner) [reply writing/freeform/shopping/qna.yaml:14]: Seems resolved now

File: writing/freeform/shopping/qna.yaml (hunk 4 above)
Comment (ckadner) [reply writing/freeform/shopping/qna.yaml:16]: Seems resolved now

File: writing/freeform/shopping/qna.yaml (hunk 1 above)
Comment (ckadner) [reply writing/freeform/shopping/qna.yaml]: Resolved now

**Review by ckadner — CHANGES_REQUESTED** [review]
This should probably be a grounded skill, similar to [foundational_skills/reasoning/logical_reasoning/general/qna.yaml](https://github.com/instructlab/taxonomy/blob/main/foundational_skills/reasoning/logical_reasoning/general/qna.yaml)? Could you move this somewhere accordingly?

Also, there are a few fields missing, like `task_description`, `seed_examples`, ... https://github.com/instructlab/taxonomy?tab=readme-ov-file#skills-yaml-examples

**Comment by mingxzhao** [comment]
@instructlab-bot precheck

**Comment by mingxzhao** [comment]
@instructlab-bot enable

**Comment by lhellebr** [comment]
@mingxzhao your precheck failed with `No modified YAML files detected in the PR for precheck`, yet there is a new yaml in my PR

**Comment by mingxzhao** [comment]
Yes apologies this was my mistake. Because the PR was placed into the foundation skills taxonomy, we are unable to run any prechecks on this. If you would like to, you could potentially move it into the extraction skills in compositional skills, which would allow me to run precheck.

**Comment by lhellebr** [comment]
@ckadner requested me to move it to foundational to writing. @mingxzhao  if you don't insist on me moving it, I'd prefer not to do any further changes and get this merged.

**Comment by bjhargrave** [comment]
> I do think this skill is in a suitable place as foundational reasoning

We are not currently accepting core skills contributions (in foundational_skills tree).

https://github.com/instructlab/community/blob/main/docs/SKILLS_GUIDE.md#core-skills

**Comment by ckadner** [comment]
> > I do think this skill is in a suitable place as foundational reasoning
> 
> We are not currently accepting core skills contributions (in foundational_skills tree).
> 
> https://github.com/instructlab/community/blob/main/docs/SKILLS_GUIDE.md#core-skills

@bjhargrave -- I made the assessment that since this skill is a foundational skill as it requires logical reasoning to compile the shopping list. @obuzek seemed to have thought otherwise in her comment https://github.com/instructlab/taxonomy/pull/330#issuecomment-2015350286 on the original PR:
> Ah excellent, an extraction skill masquerading as a recipe skill 😉

So, if it's not 100% clear cut, the question is whether this skill could be construed to be a compositional/extraction skill, we move the file and get our pipeline running (at least to the point of running pre-check).

**Comment by lhellebr** [comment]
Guys, whatever floats your boat, let me know if you need me to do something :)

**Comment by lhellebr** [comment]
@ckadner To be clear, yes, of course I'm ok with you doing that
> if you are okay with that, I could move it on your branch, run the checks and then we can see about next steps. Without our checks running on this skill, the PR won't be able to progress at this point.

**Comment by mingxzhao** [comment]
Ok am going to go ahead and move to test it. Sorry for the delay!

**Comment by mingxzhao** [comment]
Ah I see a small issue, you will still need to create an attribution.txt file to go with it. I assume this is self authored, so that is all you need to put in the file, but every submission needs some sort of attribution. Please place it in the directory I have placed this qna into.

**Comment by mingxzhao** [comment]
@instructlab-bot precheck

**Comment by mingxzhao** [comment]
Ok so based on the precheck results, the model seems fairly competent at completing the task, so I will close this PR. I apologize for the long process this took, we are still working on ironing out some issues, but thank you so much for your contribution!

**Pull request closed without merging**